File: LeetCode/Greedy/JS/Leetcode2038-移除相邻颜色.js
var winnerOfGame = function(colors) {
    let cur = 'C',f = [0,0],cnt = 1;
    for(const c of colors){
        if(cur != c){
            cur = c;
            cnt = 1
        }else{
            cnt++
            if(cnt >= 3) f[cur.charCodeAt(0) - 'A'.charCodeAt(0)]++;
        }
    }
    return f[0] > f[1];
};

File: LeetCode/Greedy/Python/Leetcode2038-移除相邻颜色.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        f, cur, cnt = [0, 0], 'C', 1
        for c in colors:
            if c != cur:
                cur = c
                cnt = 1
            else:
                cnt += 1
                if cnt >= 3:
                    f[ord(cur) - ord('A')] += 1
        return f[0] > f[1]
